### DISCUSSION TITLE
How to use with Hyper HDR

### DISCUSSION THREAD

**Discussion by EvoSems** [awawa-dev/HyperSerialEsp8266#5] [General]
This seems a great bit of software! thanks for all the hard work so far!

Could someone explain simply to someone like myself how I use this with hyperhdr?

I have hyperhdr running on a pi for an ambilight effect. I also have a few esp8266. I have flashed one with this software easily.

is it my understanding that I somehow connect the esp to the pi somehow to take over the output? would someone mind explaining a simple instruction for someone like me? 

many many thanks in advance.

**Comment by awawa-dev** [comment]
HI
Exactly, HyperHDR outputs data through USB to the connected ESP8266 device at very high speed. HyperSerialEsp8266 running on the ESP8266 receives it and commands the LED strip to update. Your ESP8266 must have 2Mb speed capable chip (like CH340G or CP2104) on the board...you must verify it with your eyes, otherwise it won't work. On the main page of HyperSerialEsp8266 github homepage there is a screenshot with the configuration of HyperHDR: it's simply, only 2 options. HyperSerialEsp8266 doesn't need a configuration, only valid firmware fits your LED strip from the release page.

Regards
Awawa

**Comment by EvoSems** [comment]
Thankyou Very much!

**Comment by EvoSems** [comment]
Hi again. hope you don't mind me commenting here. its unrelated to the above.

my setup is working great with HyperserialWLED @ AWA 2000000 = ) (Very Happy)

I see you have provided HyperSPI project = ) (Thankyou!)

I assume this will be better than AWA with usb. would you recommend using this with ezcap320? what performance and features would be upgraded from AWA? 

P.s Which RPI GPIO pins exactly will be used? (sk6812)

Many thanks

**Comment by awawa-dev** [comment]
Hi
As you can see in the HyperSPI I can't make my Ezcap 320 to work in USB3.0 mode on my Rpi4. Have some other feedback from users that some managed to make it work (probably depends on the Ezcap revision) but it doesn't work stable so they switch it to the USB2.0 port anyway. So if you have this grabber I think it's worth to try.  
For a performance: we drop all the serial buffer over USB handling on both sides, SPI is extremely light. I think the only obstacle could be your setup: if you can afford to put 3 short cables from Rpi to esp8266...if your esp is located further that can be a problem. There are two other factors that favors HyperSPI: it's recommended for larger number of LED because serial buffer could be insufficient to transfer larger data in time or if you want to try some unique modes that can not be achieved with WLED or HyperSerialEsp8266 like 40Mhz data transfer for HD107 LED strip (but it requires Esp32).  
if you choose default SPI config (spidev0.0) then RPi : SCK=11, MISO=9, MOSI=10 (should be visible on the screenshot on HyperSPI)
Before jumping to HyperSPI try first HyperSerialEsp8266 if the RGB=>RGBW algorithm fits you: it' different than WLED one.

**Comment by EvoSems** [comment]
I see. well more performance the better I'd say! 
 my 320 is coming from china currently. so probably won't see it for a few weeks... my only problem for me is my pi isn't 100%
Testing...
Write 0 to gpio 10 failed.
Pull down on gpio 10 failed.
Write 1 to gpio 18 failed.
Pull up on gpio 18 failed.

I'm assuming it would use these pins...?

I will give HyperSerialEsp8266 a try, is the difference mainly a passthrough device vs a wifi front end with effects + passthrough?

many thanks

**Comment by awawa-dev** [comment]
WiringPi is a bit deprecated especially for Rpi4. Don't know if it's still reliable test. Dead SPI0 MOSI is a problem...Rpi has other SPI1 interface 19,20,21 (CS1: gpio 18 is not required for HyperSPI and esp8266)  but you must enable it in the raspbian first and reboot.

HyperSerialEsp8266 vs HyperSerialWLED: lower power usage, pure neopixelbus library when we have a full control and we choose pins that should be used (in some case not these wanted by the user ;-) ), no WLED interruption for the serial port transmission, CONS: you loose WLED interface and all its stuff.

**Reply by EvoSems** [reply]
ah ok. so some home with damaged pi. I will look into enable spi1. thanks

flashed HyperSerialEsp8266 just now. much more responsive. my only problem is with white static light (6500k) the led strip varies in colour (red tinge) I fixed this in WLED with auto voltage (850ma) maybe my power supply 5v 10a / 260 sk8612 NW isn't powerful enough? I have powered from both ends of the led strip. is there any setting in HyperHDR to lower the voltage/brightness? many thanks again.

**Reply by awawa-dev** [reply]
You can lower the brightness/lower max limit in the processing tab. For similar 270RGBW I use 100W (20A) power supply in my setup.

**Reply by EvoSems** [reply]
awesome thank you. I think I will beef up the power supply as I want a hassle free setup = ) my tv is a massive pain to remove from the wall...

edit. need to upgrade power. 60% limit is with no tinge to white.

p.s I found a home bridge plugin for Hyperion. I can now change brightness and turn on and off HyperHDR via Homekit on my iOS devices = )

**Comment by EvoSems** [comment]
Just wanted to let you know my setup is complete. Thank you very much for your work on this project! I finally have 4K HDR ambilight with working CEC

BRAVIA 65” OLED
APPLE TV HDR 4K
Ezcap 320
RPi 4
HyperHDR 17
HyperSPI
CEC Working with Ali HDMI Splitter

**Reply by catweazle0172** [reply]
Hello EvoSems,

I am also busy to setup up almost the same configuration with an Apple TV 4K. But I am now struggling with dhcp protection errors on the Escap 320.
Can you forward me your type of hdmi splitter that you are using?
Thank you!
Greetings,
Mike

**Reply by tuba59** [reply]
Hey,
I'm also curious what splitter are you using 😃 
Mine has a few hiccups, also a aliexpress one (with 2 inputs, one for Shield, one for PS5)

**Reply by catweazle0172** [reply]
Hello Tuba,

Aim now using the Ezcoo EZ-SP12HAS. 

https://www.easycoolav.com/products/4k60-hdmi-splitter-1-in-2-outdolby-vision-optical-audio-breakout

This one is working for me at 4K 60hrz

Greetings,
Mike

**Reply by ForxPorx** [reply]
Using the HDMI Switch that mirrors 8K 60Hz HMI ports is the best solution then you can use the cheapest grabber USB 2.0 low resolution at 60Hz - it is enough for HyperHDR and more than enough for Hyperion.  HyperSerial or HyperSPI will make sure LED output is 60+ Hz
